lib/app/components/auth/passkey_prompt_dialog.dart

```diff
@@ -7,13 +7,31 @@ import 'package:ion/app/components/progress_bar/ion_loading_indicator.dart';
 import 'package:ion/app/components/screen_offset/screen_bottom_offset.dart';
 import 'package:ion/app/extensions/extensions.dart';
 import 'package:ion/app/features/core/providers/theme_mode_provider.dart';
+import 'package:ion/app/hooks/use_on_init.dart';
 import 'package:ion/generated/assets.gen.dart';
 
-class PasskeysPromptDialog extends ConsumerWidget {
-  const PasskeysPromptDialog({super.key});
+class PasskeyPromptDialog extends HookConsumerWidget {
+  const PasskeyPromptDialog({
+    required this.onInit,
+    required this.provider,
```

Comment:
don't pass a provider as a widget parameter (https://riverpod.dev/docs/essentials/do_dont). Use `Ref` or pass a callback, or use a riverpod-provider directly in `ref.listen`

Reply:
discussed